agent/engine/host_resource_manager.go

```diff
@@ -16,20 +16,226 @@
 package engine
 
 import (
+	"fmt"
+	"sync"
+
 	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
 	"github.com/aws/amazon-ecs-agent/agent/logger"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 )
 
+// TODO remove this once resource, consume are used
+//lint:file-ignore U1000 Ignore all unused code
+
 // initialHostResource keeps account of each task in
 type HostResourceManager struct {
-	initialHostResource map[string]*ecs.Resource
-	consumedResource    map[string]*ecs.Resource
+	initialHostResource       map[string]*ecs.Resource
+	consumedResource          map[string]*ecs.Resource
+	hostResourceManagerRWLock sync.Mutex
 
 	//task.arn to boolean whether host resources consumed or not
 	taskConsumed map[string]bool
 }
 
+// Returns if resources consumed or not and error status
+// false, nil -> did not consume, should stay pending
+// false, err -> resources map has errors, something went wrong
+// true, nil -> successfully consumed
+func (h *HostResourceManager) consume(taskArn string, resources map[string]*ecs.Resource) (bool, error) {
```

Comment:
I see these functions to be used only in engine package. Unless there are good reasons to export them, I believe we can keep them as it is

Reply:
Why don't we put host resource manager in its own package? How are consumers of host resource manager supposed to know what methods form its public API?

Reply:
We put it in engine (and not in it's own package) because we only see `HostResourceManager` as being used to enable task scheduling, specifically in `task_manager` and `docker_task_engine`, and hence did not export the methods as well.